### PR TITLE
Implement storybook and create reusable components

### PR DESCRIPTION
**Pull request by RubenWerdmuller**

### CODE REVIEW

File: src/components/client-add/ClientAdd.jsx [hunk 1]
```diff
@@ -58,104 +61,79 @@ const ClientAdd = ({ addClient, toggleFormVisibility }) => {
...
+          <Label>
+            <span>Branch</span>
+            <Input
+              isValid={validity.branch}
               maxLength="30"
               minLength="2"
               name="branch"
               onBlur={handleBlur}
-              onChange={({ target }) => setBranch(target.value)}
+              onChange={({ target }) => setCompanyName(target.value)}
```
Comment (dylanjongbloed) [inline src/components/client-add/ClientAdd.jsx]: shouldn't this remain setBranch?

File: src/components/client-add/ClientAdd.jsx [hunk 2]
```diff
@@ -58,104 +61,79 @@ const ClientAdd = ({ addClient, toggleFormVisibility }) => {
...
         </div>
         <span className={styles.verticalLine} />
         <div className={styles.inputContainer}>
-          <label className={`${styles.label}`} htmlFor="branchAdress">
-            <span className={styles.labelText}>Branch adress</span>
-            <input
-              className={`${styles.input} 
-              ${validity.branchAdress === false && styles.invalidInput}`}
+          <Label>
+            Branch adress
```
Comment (dylanjongbloed) [inline src/components/client-add/ClientAdd.jsx]: Address is spelled with two d's

File: shared/components/input/Input.jsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+const Input = styled.input`
+  background-image: #fff;
+  border: solid 1px #ced0da;
+  border-color: ${props => props.isValid === false && '#fb6375'};
```
Comment (RobbinHabermehl) [inline shared/components/input/Input.jsx]: What value will `border-color` have if it evaluates to `true`? Also, the `isValid` prop should be a boolean anyway.

File: shared/components/label/Label.jsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  @media only screen and (min-width: 901px) {
+    margin-bottom: 20px;
+
+    &:last-child {
+      margin: 0;
+    }
+  }
+
+  @media only screen and (min-width: 901px) {
+    margin: ${props => props.halfWidth && '0'};
```
Comment (RobbinHabermehl) [inline shared/components/label/Label.jsx]: What happens if `halfWidth` is `false`?

File: src/services/team-member-api.js [hunk 5]
```diff
@@ -1,4 +1,8 @@
-const apiUrl = 'http://localhost:3000/team-members';
+const apiUrl = `${
```
Comment (RobbinHabermehl) [inline src/services/team-member-api.js]: You're not restoring the server-side sorting?

File: shared/components/button/Button.jsx [hunk 6]
```diff
@@ -0,0 +1,46 @@
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+const Button = styled.button`
+  background-color: ${props =>
+    props.disabled === false ? '#39b54a' : '#f5f7f9'};
+  border-radius: 4px;
+  border: ${props => (props.disabled === false ? '0' : '1px solid #ced0da')};
+  color: ${props => (props.disabled === false ? '#fff' : '#4b5464')};
```
[the remark this answers is not in the document]
Comment (RubenWerdmuller) [reply shared/components/button/Button.jsx]: In this case I use 3 states to regulate the prop disabled. `true`, `false` and `undefined`.

File: shared/components/button/Button.jsx [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  fill: #a8aab7;
+  font-size: 14px;
+  height: 36px;
+  outline: 0;
+  padding: 0;
+  text-align: center;
+  width: calc(50% - 5px);
+
+  & + & {
```
[the remark this answers is not in the document]
Comment (RubenWerdmuller) [reply shared/components/button/Button.jsx]: It does!

File: shared/components/input/Input.jsx (hunk 3 above)
Comment (RubenWerdmuller) [reply shared/components/input/Input.jsx]: I use 3 different states to regulate the prop isValid. true, false and undefined.

File: shared/components/label/Label.jsx (hunk 4 above)
Comment (RubenWerdmuller) [reply shared/components/label/Label.jsx]: When `halfWidth` is `false`, the label is full width on desktop.

File: src/services/team-member-api.js (hunk 5 above)
Comment (RubenWerdmuller) [reply src/services/team-member-api.js]: This happens specifically under `fetchTeamMembers`

File: shared/components/button/Button.jsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (RubenWerdmuller) [reply shared/components/button/Button.jsx]: It does!

File: shared/components/button/Button.jsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (RobbinHabermehl) [reply shared/components/button/Button.jsx]: It's preferable to just have `true` or `false`. If really necessary have the component it's being used in evaluate it to a boolean.

File: shared/components/button/Button.jsx [hunk 8]
```diff
@@ -4,8 +4,9 @@ import styled from 'styled-components';
 const Button = styled.button`
   background-color: ${props =>
     props.disabled === false ? '#39b54a' : '#f5f7f9'};
+  border: 1px solid #ced0da;
   border-radius: 4px;
-  border: ${props => (props.disabled === false ? '0' : '1px solid #ced0da')};
+  border-width: ${props => (props.disabled === false ? '0' : '1px')};
```
Comment (RobbinHabermehl) [inline shared/components/button/Button.jsx]: Now you're actually introducing more code. It wasn't intended to have you change this, but what I meant was:
`border: ${props => !props.disabled ? 0 : '1px'} solid #ced0da;`

File: shared/components/input/Input.jsx [hunk 9]
```diff
@@ -4,7 +4,7 @@ import styled from 'styled-components';
 const Input = styled.input`
   background-image: #fff;
   border: solid 1px #ced0da;
-  border-color: ${props => props.isValid === false && '#fb6375'};
+  border-color: ${props => (props.isValid === false ? '#fb6375' : '#ced0da')};
```
Comment (RobbinHabermehl) [inline shared/components/input/Input.jsx]: Write as short-hand.

File: shared/components/label/Label.jsx (hunk 4 above)
Comment (RobbinHabermehl) [reply shared/components/label/Label.jsx]: And what is the value of the property `width` then?

File: shared/components/button/Button.jsx (hunk 8 above)
Comment (RubenWerdmuller) [reply shared/components/button/Button.jsx]: Nice solution!

**Pull request merged**